app/src/main/kotlin/com/mapbox/vision/teaser/replayer/SessionsAdapter.kt

```diff
@@ -1,42 +1,180 @@
+            }
+            holder.iconChecked.visibility = INVISIBLE
+        }
+    }
+
+    private fun bindViewHolderClickListener(holder: SessionViewHolder, item: AdapterItem) {
+        if (!(item is CameraItem && isMultiSelection)) {
+            holder.itemView.setOnClickListener {
+                if (isMultiSelection) {
+                    handleMultiSelectionClick(holder, item)
+                }
+                if (item is CameraItem) {
+                    clickCameraListener.invoke()
+                } else {
+                    clickSessionListener.invoke(item.name)
+                }
+            }
+        }
+    }
+
+    private fun handleMultiSelectionClick(holder: SessionViewHolder, item: AdapterItem) {
+        if (selectedItems.contains(item.name)) {
+            holder.itemView.setBackgroundColor(backgroundColorTransparent)
+            holder.iconChecked.setImageResource(R.drawable.ic_not_checked)
+            selectedItems.remove(item.name)
+        } else {
+            holder.itemView.setBackgroundColor(backgroundColorSelection)
+            holder.iconChecked.setImageResource(R.drawable.ic_checked)
+            selectedItems.add(item.name)
+        }
+    }
+
+    private fun bindViewHolderLongClick(holder: SessionViewHolder, item: AdapterItem) {
+        if (item !is CameraItem) {
+            holder.itemView.setOnLongClickListener {
+                if (isMultiSelection) {
+                   handleMultiSelectionClick(holder, item)
+                } else {
+                    selectedItems.add(item.name)
+                    onActivateMultiSelectionListener.invoke()
+                }
+                true
+            }
+        }
+    }
 
-        init {
-            textView.setOnClickListener {
-                clickListener.invoke(sessions[adapterPosition])
+    fun activateMultiSelection() {
+        isMultiSelection = true
+        notifyDataSetChanged()
+    }
+
+    fun resetMultiSelection() {
+        isMultiSelection = false
+        selectedItems.clear()
+        notifyDataSetChanged()
+    }
+
+    fun selectAll() {
+        items.forEach {
+            if (it is SessionItem) {
+                selectedItems.add(it.name)
             }
         }
+        notifyDataSetChanged()
+    }
+
+    fun updateSessionsList() {
+        items.clear()
+        items.add(CameraItem(cameraString))
+        File(basePath).listFiles().forEach {
+            val dateString = dateFormatter.format(it.lastModified())
+            items.add(SessionItem(it.name, dateString))
+        }
+    }
+
+    override fun getItemCount() = items.size
+
+    inner class SessionViewHolder(view: View) : RecyclerView.ViewHolder(view) {
+        val textSessionName: TextView = view.findViewById(R.id.session_view_session_name)
+        val textSessionDate: TextView = view.findViewById(R.id.session_view_session_date)
+        val iconChecked: ImageView = view.findViewById(R.id.session_view_checked)
+    }
+
+    companion object {
```

Comment:
AFAIR we put companion object block on top of the file

Reply:
fixed